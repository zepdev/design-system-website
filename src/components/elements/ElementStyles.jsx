import React from 'react'
import PropTypes from 'prop-types'
import Headline from '../typography/Headline'
import Table from '../../components/table/Table.jsx'
import Structure from '../../components/structure/Structure.jsx'
import elements from '../../data/elements.js'

const ElementStyles = ({ element }) => {
  return (
    <>
      <Headline variant="sm">Typography</Headline>

      <Table
        title="typography"
        content={elements[element].typography}
        header={['name', 'px', 'rem', 'specifications', 'actual size']}
      />

      <Headline variant="sm">Colors</Headline>

      <Table
        title="typography"
        content={elements[element].colors}
        header={['name', 'property', 'zeppelin', 'cat', 'rental']}
      />
      <Headline variant="sm">Structure</Headline>

      <Structure element="button" />

      <Table
        title="structure"
        content={elements[element].structure}
        header={['name', 'property', 'px', 'rem']}
      />
    </>
  )
}

ElementStyles.propTypes = {
  element: PropTypes.string.isRequired,
}

export default ElementStyles
