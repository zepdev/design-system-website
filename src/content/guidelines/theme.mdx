---
label: Guidelines
title: Theme
---

# Theme

Themes are used to customize existing elements to fit the specific style of a strategic business unit or product. There is no base color for the Zeppelin-element-library, almost all elements require that a theme class is present. At the moment the Zeppelin Design System offers three main themes. Developers can customize all elements by adding the theme class to any html tag (`<main>`,`<body>`,`<div>`, ...) as long it wraps all elements you want to use.

```js
<main class="theme-zeppelin" id="app-root">
  ...
</main>
```

We currently offer three main themes;

```js
// Zeppelin Indigo Blue
<div class="theme-zeppelin" >...</div>

// CAT Yellow
<div class="theme-cat" >...</div>

// Rental Red
<div class="theme-rental" >...</div>
```
