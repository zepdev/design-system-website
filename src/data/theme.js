module.exports = {
  breakpoints: {
    xs: '36em' /* 576px */,
    s: '48em' /* 768px */,
    m: '62em' /* 992px */,
    l: '75em' /* 1200px */,
    xl: '105em' /* 1680px */,
    xxl: '120em' /* 1920px */,
  },
  typography: {
    typefaces: {
      roboto: "'Roboto', Arial, Helvetica, sans-serif",
      teko: "Teko', sans-serif",
    },
    basics: {
      fontSizeBase: {
        specification: 'font-size-base',
        px: '16px',
        rem: '1rem',
        weight: null,
      },
      lineHeightBase: {
        specification: 'line-height-base',
        px: '24px',
        rem: '1.5rem',
        weight: null,
      },
      fontWeightBase: {
        specification: 'font-weight-base',
        px: null,
        rem: null,
        weight: 400,
      },
    },
    headings: {
      normal: {
        'zep-typo--normal-1': {
          rem: '0.75rem',
          px: '12px',
          weight: 400,
          specifications: [
            'font-family: "Roboto"',
            'font-weight: 400',
            'line-height: 1.5',
            'text-transform: "none"',
          ],
        },
        'zep-typo--normal-2': {
          rem: '0.75rem',
          px: '12px',
          weight: 400,
          specifications: [
            'font-family: "Roboto"',
            'font-weight: 400',
            'line-height: 1.5',
            'text-transform: "none"',
          ],
        },
        'zep-typo--normal-3': {
          rem: '0.75rem',
          px: '12px',
          weight: 400,
          specifications: [
            'font-family: "Roboto"',
            'font-weight: 400',
            'line-height: 1.5',
            'text-transform: "none"',
          ],
        },
        'zep-typo--normal-4': {
          rem: '0.75rem',
          px: '12px',
          weight: 400,
          specifications: [
            'font-family: "Roboto"',
            'font-weight: 400',
            'line-height: 1.5',
            'text-transform: "none"',
          ],
        },
        'zep-typo--normal-5': {
          rem: '0.75rem',
          px: '12px',
          weight: 400,
          specifications: [
            'font-family: "Roboto"',
            'font-weight: 400',
            'line-height: 1.5',
            'text-transform: "none"',
          ],
        },
        'zep-typo--normal-6': {
          rem: '0.75rem',
          px: '12px',
          weight: 400,
          specifications: [
            'font-family: "Roboto"',
            'font-weight: 400',
            'line-height: 1.5',
            'text-transform: "none"',
          ],
        },
        'zep-typo--normal-7': {
          rem: '0.75rem',
          px: '12px',
          weight: 400,
          specifications: [
            'font-family: "Roboto"',
            'font-weight: 400',
            'line-height: 1.5',
            'text-transform: "none"',
          ],
        },
        'zep-typo--normal-8': {
          rem: '0.75rem',
          px: '12px',
          weight: 400,
          specifications: [
            'font-family: "Roboto"',
            'font-weight: 400',
            'line-height: 1.5',
            'text-transform: "none"',
          ],
        },
      },
      display: {
        'zep-typo--display-1': {
          rem: '0.75rem',
          px: '12px',
          weight: 400,
          specifications: [
            'font-family: "Roboto"',
            'font-weight: 400',
            'line-height: 1.5',
            'text-transform: "none"',
          ],
        },
        'zep-typo--display-2': {
          rem: '0.75rem',
          px: '12px',
          weight: 400,
          specifications: [
            'font-family: "Roboto"',
            'font-weight: 400',
            'line-height: 1.5',
            'text-transform: "none"',
          ],
        },
        'zep-typo--display-3': {
          rem: '0.75rem',
          px: '12px',
          weight: 400,
          specifications: [
            'font-family: "Roboto"',
            'font-weight: 400',
            'line-height: 1.5',
            'text-transform: "none"',
          ],
        },
        'zep-typo--display-4': {
          rem: '0.75rem',
          px: '12px',
          weight: 400,
          specifications: [
            'font-family: "Roboto"',
            'font-weight: 400',
            'line-height: 1.5',
            'text-transform: "none"',
          ],
        },
      },
      special: {
        'zep-typo--special-1': {
          rem: '0.75rem',
          px: '12px',
          weight: 400,
          specifications: [
            'font-family: "Roboto"',
            'font-weight: 400',
            'line-height: 1.5',
            'text-transform: "none"',
          ],
        },
        'zep-typo--special-2': {
          rem: '0.75rem',
          px: '12px',
          weight: 400,
          specifications: [
            'font-family: "Roboto"',
            'font-weight: 400',
            'line-height: 1.5',
            'text-transform: "none"',
          ],
        },
        'zep-typo--special-3': {
          rem: '0.75rem',
          px: '12px',
          weight: 400,
          specifications: [
            'font-family: "Roboto"',
            'font-weight: 400',
            'line-height: 1.5',
            'text-transform: "none"',
          ],
        },
        'zep-typo--special-4': {
          rem: '0.75rem',
          px: '12px',
          weight: 400,
          specifications: [
            'font-family: "Roboto"',
            'font-weight: 400',
            'line-height: 1.5',
            'text-transform: "none"',
          ],
        },
      },
    },
  },
  colors: {
    primary: {
      indigoBlue: {
        name: 'indigo-blue',
        hex: '#27166f',
        rgb: '39/22/111',
        hsb: null,
      },
      catYellow: {
        name: 'cat-yellow',
        hex: '#f0b823',
        rgb: '240/184/35',
        hsb: null,
      },
      rentalRed: {
        name: 'rental-red',
        hex: '#af0e0e',
        rgb: '175/14/14',
        hsb: null,
      },
    },
    gray: {
      grayDark: {
        name: 'gray-dark',
        hex: '#101111',
        rgb: '16/17/17',
        hsb: null,
      },
      grayMidDark: {
        name: 'gray-mid-dark',
        hex: '#373d3f',
        rgb: '55/61/63',
        hsb: null,
      },
      gray: { name: 'gray', hex: '#464a4c', rgb: '70/74/76', hsb: null },
      grayLight: {
        name: 'gray-light',
        hex: '#6f7c80',
        rgb: '111/124/128',
        hsb: null,
      },
      grayLighter: {
        name: 'gray-lighter',
        hex: '#eceeef',
        rgb: '236/238/239',
        hsb: null,
      },
      grayMid: {
        name: 'gray-mid',
        hex: '#f2f3f4',
        rgb: '242/243/244',
        hsb: null,
      },
      grayLightest: {
        name: 'gray-lightest',
        hex: '#f7f7f9',
        rgb: '247/247/249',
        hsb: null,
      },
      white: { name: 'white', hex: '#ffffff', rgb: '255/255/255', hsb: null },
    },
    support: {
      alert: { name: 'alert', hex: '#ff6562', rgb: '255/101/98', hsb: null },
      warning: {
        name: 'warning',
        hex: '#f0ad4e',
        rgb: '240,/73/78',
        hsb: null,
      },
      success: { name: 'success', hex: '#5cb85c', rgb: '92/184/92', hsb: null },
      info: { name: 'info', hex: '#5bc0de', rgb: '91/192/222', hsb: null },
    },
  },
  spacing: {
    xxs: 4,
    xs: 8,
    s: 12,
    m: 16,
    l: 24,
    xl: 32,
    xxl: 48,
  },
}
