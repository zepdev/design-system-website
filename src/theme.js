import { createMuiTheme } from '@material-ui/core/styles'
import guidelines from '@zlab-de/zel-react/guidelines.json'

const theme = createMuiTheme({
  palette: {
    type: 'light',
    primary: {
      light: guidelines.theme.indigo.secondary,
      main: guidelines.theme.indigo.primary,
      dark: guidelines.theme.indigo.primary,
    },
    error: { main: guidelines.color.support.alert },
  },
  status: {
    gray: '#eceeef',
  },
  logo: guidelines.logo,
  blue: guidelines.theme.blue,
  indigo: guidelines.theme.indigo,
  red: guidelines.theme.red,
  yellow: guidelines.theme.yellow,
  guidelines: guidelines,
  space: guidelines.spacing.component,
  color: guidelines.color,
})

export default theme
