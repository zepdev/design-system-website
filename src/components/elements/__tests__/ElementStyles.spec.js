import React from 'react'
import { render } from 'react-testing-library'
import { ThemeProvider } from 'react-jss'
import theme from 'zeppelin-element-library/bundle/themes/theme.json'
import ElementStyles from '../ElementStyles'

describe('ElementStyles', () => {
  it('renders correctly', () => {
    const mockProps = {
      element: 'button',
    }
    const component = render(
      <ThemeProvider theme={theme}>
        <ElementStyles {...mockProps} />
      </ThemeProvider>
    )

    expect(component).toMatchSnapshot()
  })
})
