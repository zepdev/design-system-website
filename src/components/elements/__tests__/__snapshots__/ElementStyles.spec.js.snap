// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ElementStyles renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <p
        class="Headline-text-0-1-1 zep-typo--normal-4"
      >
        Typography
      </p>
      <div
        class="Table-root-0-1-2"
      >
        <table
          class="Table-table-0-1-3"
        >
          <caption
            class="zep-visually-hidden"
          >
            typography
          </caption>
          <thead
            class="Table-header-0-1-7"
          >
            <tr
              class="Table-tr-0-1-5"
            >
              <th
                class="Table-th-0-1-4 zep-typo--normal-1 Table-capitalize-0-1-9"
                scope="col"
              >
                name
              </th>
              <th
                class="Table-th-0-1-4 zep-typo--normal-1 Table-lowercase-0-1-8"
                scope="col"
              >
                px
              </th>
              <th
                class="Table-th-0-1-4 zep-typo--normal-1 Table-lowercase-0-1-8"
                scope="col"
              >
                rem
              </th>
              <th
                class="Table-th-0-1-4 zep-typo--normal-1 Table-capitalize-0-1-9"
                scope="col"
              >
                specifications
              </th>
              <th
                class="Table-th-0-1-4 zep-typo--normal-1 Table-capitalize-0-1-9"
                scope="col"
              >
                actual size
              </th>
            </tr>
          </thead>
          <tbody
            class="Table-tbody-0-1-6"
          >
            <tr
              class="Table-tr-0-1-5"
            >
              <td
                class="Table-th-0-1-4 Table-borderBottom-0-1-10 zep-typo--normal-2"
                scope="row"
              >
                zep-typo--special-4
              </td>
              <td
                class="Table-th-0-1-4 Table-borderBottom-0-1-10 zep-typo--normal-2"
              >
                24px
              </td>
              <td
                class="Table-th-0-1-4 Table-borderBottom-0-1-10 zep-typo--normal-2"
              >
                1.5rem
              </td>
              <td
                class="Table-th-0-1-4 Table-borderBottom-0-1-10 zep-typo--normal-2"
              >
                <span>
                  <p
                    class="zep-typo--normal-2"
                  >
                    font-family: "Teko"
                  </p>
                  <p
                    class="zep-typo--normal-2"
                  >
                    font-weight: 400 (regular)
                  </p>
                  <p
                    class="zep-typo--normal-2"
                  >
                    line-height: 1.5 (24px)
                  </p>
                  <p
                    class="zep-typo--normal-2"
                  >
                    text-transform: uppercase
                  </p>
                </span>
              </td>
              <td
                class="Table-th-0-1-4 Table-borderBottom-0-1-10 zep-typo--special-4"
              >
                lorem ipsum
              </td>
            </tr>
          </tbody>
        </table>
      </div>
      <p
        class="Headline-text-0-1-1 zep-typo--normal-4"
      >
        Colors
      </p>
      <div
        class="Table-root-0-1-2"
      >
        <table
          class="Table-table-0-1-3"
        >
          <caption
            class="zep-visually-hidden"
          >
            typography
          </caption>
          <thead
            class="Table-header-0-1-7"
          >
            <tr
              class="Table-tr-0-1-5"
            >
              <th
                class="Table-th-0-1-4 zep-typo--normal-1 Table-capitalize-0-1-9"
                scope="col"
              >
                name
              </th>
              <th
                class="Table-th-0-1-4 zep-typo--normal-1 Table-capitalize-0-1-9"
                scope="col"
              >
                property
              </th>
              <th
                class="Table-th-0-1-4 zep-typo--normal-1 Table-capitalize-0-1-9"
                scope="col"
              >
                zeppelin
              </th>
              <th
                class="Table-th-0-1-4 zep-typo--normal-1 Table-capitalize-0-1-9"
                scope="col"
              >
                cat
              </th>
              <th
                class="Table-th-0-1-4 zep-typo--normal-1 Table-capitalize-0-1-9"
                scope="col"
              >
                rental
              </th>
            </tr>
          </thead>
        </table>
      </div>
      <p
        class="Headline-text-0-1-1 zep-typo--normal-4"
      >
        Structure
      </p>
      <div
        class="Structure-root-0-1-11"
      >
        <img
          src="test-file-stub"
        />
      </div>
      <div
        class="Table-root-0-1-2"
      >
        <table
          class="Table-table-0-1-3"
        >
          <caption
            class="zep-visually-hidden"
          >
            structure
          </caption>
          <thead
            class="Table-header-0-1-7"
          >
            <tr
              class="Table-tr-0-1-5"
            >
              <th
                class="Table-th-0-1-4 zep-typo--normal-1 Table-capitalize-0-1-9"
                scope="col"
              >
                name
              </th>
              <th
                class="Table-th-0-1-4 zep-typo--normal-1 Table-capitalize-0-1-9"
                scope="col"
              >
                property
              </th>
              <th
                class="Table-th-0-1-4 zep-typo--normal-1 Table-lowercase-0-1-8"
                scope="col"
              >
                px
              </th>
              <th
                class="Table-th-0-1-4 zep-typo--normal-1 Table-lowercase-0-1-8"
                scope="col"
              >
                rem
              </th>
            </tr>
          </thead>
          <tbody
            class="Table-tbody-0-1-6"
          >
            <tr
              class="Table-tr-0-1-5"
            >
              <td
                class="Table-th-0-1-4 Table-borderBottom-0-1-10 zep-typo--normal-2"
                scope="row"
              >
                btn-height
              </td>
              <td
                class="Table-th-0-1-4 Table-borderBottom-0-1-10 zep-typo--normal-2"
              >
                height
              </td>
              <td
                class="Table-th-0-1-4 Table-borderBottom-0-1-10 zep-typo--normal-2"
              >
                48px
              </td>
              <td
                class="Table-th-0-1-4 Table-borderBottom-0-1-10 zep-typo--normal-2"
              >
                3rem
              </td>
            </tr>
            <tr
              class="Table-tr-0-1-5"
            >
              <td
                class="Table-th-0-1-4 Table-borderBottom-0-1-10 zep-typo--normal-2"
                scope="row"
              >
                btn-padding
              </td>
              <td
                class="Table-th-0-1-4 Table-borderBottom-0-1-10 zep-typo--normal-2"
              >
                padding
              </td>
              <td
                class="Table-th-0-1-4 Table-borderBottom-0-1-10 zep-typo--normal-2"
              >
                16px, 18px
              </td>
              <td
                class="Table-th-0-1-4 Table-borderBottom-0-1-10 zep-typo--normal-2"
              >
                1rem, 1.125rem
              </td>
            </tr>
          </tbody>
        </table>
      </div>
    </div>
  </body>,
  "container": <div>
    <p
      class="Headline-text-0-1-1 zep-typo--normal-4"
    >
      Typography
    </p>
    <div
      class="Table-root-0-1-2"
    >
      <table
        class="Table-table-0-1-3"
      >
        <caption
          class="zep-visually-hidden"
        >
          typography
        </caption>
        <thead
          class="Table-header-0-1-7"
        >
          <tr
            class="Table-tr-0-1-5"
          >
            <th
              class="Table-th-0-1-4 zep-typo--normal-1 Table-capitalize-0-1-9"
              scope="col"
            >
              name
            </th>
            <th
              class="Table-th-0-1-4 zep-typo--normal-1 Table-lowercase-0-1-8"
              scope="col"
            >
              px
            </th>
            <th
              class="Table-th-0-1-4 zep-typo--normal-1 Table-lowercase-0-1-8"
              scope="col"
            >
              rem
            </th>
            <th
              class="Table-th-0-1-4 zep-typo--normal-1 Table-capitalize-0-1-9"
              scope="col"
            >
              specifications
            </th>
            <th
              class="Table-th-0-1-4 zep-typo--normal-1 Table-capitalize-0-1-9"
              scope="col"
            >
              actual size
            </th>
          </tr>
        </thead>
        <tbody
          class="Table-tbody-0-1-6"
        >
          <tr
            class="Table-tr-0-1-5"
          >
            <td
              class="Table-th-0-1-4 Table-borderBottom-0-1-10 zep-typo--normal-2"
              scope="row"
            >
              zep-typo--special-4
            </td>
            <td
              class="Table-th-0-1-4 Table-borderBottom-0-1-10 zep-typo--normal-2"
            >
              24px
            </td>
            <td
              class="Table-th-0-1-4 Table-borderBottom-0-1-10 zep-typo--normal-2"
            >
              1.5rem
            </td>
            <td
              class="Table-th-0-1-4 Table-borderBottom-0-1-10 zep-typo--normal-2"
            >
              <span>
                <p
                  class="zep-typo--normal-2"
                >
                  font-family: "Teko"
                </p>
                <p
                  class="zep-typo--normal-2"
                >
                  font-weight: 400 (regular)
                </p>
                <p
                  class="zep-typo--normal-2"
                >
                  line-height: 1.5 (24px)
                </p>
                <p
                  class="zep-typo--normal-2"
                >
                  text-transform: uppercase
                </p>
              </span>
            </td>
            <td
              class="Table-th-0-1-4 Table-borderBottom-0-1-10 zep-typo--special-4"
            >
              lorem ipsum
            </td>
          </tr>
        </tbody>
      </table>
    </div>
    <p
      class="Headline-text-0-1-1 zep-typo--normal-4"
    >
      Colors
    </p>
    <div
      class="Table-root-0-1-2"
    >
      <table
        class="Table-table-0-1-3"
      >
        <caption
          class="zep-visually-hidden"
        >
          typography
        </caption>
        <thead
          class="Table-header-0-1-7"
        >
          <tr
            class="Table-tr-0-1-5"
          >
            <th
              class="Table-th-0-1-4 zep-typo--normal-1 Table-capitalize-0-1-9"
              scope="col"
            >
              name
            </th>
            <th
              class="Table-th-0-1-4 zep-typo--normal-1 Table-capitalize-0-1-9"
              scope="col"
            >
              property
            </th>
            <th
              class="Table-th-0-1-4 zep-typo--normal-1 Table-capitalize-0-1-9"
              scope="col"
            >
              zeppelin
            </th>
            <th
              class="Table-th-0-1-4 zep-typo--normal-1 Table-capitalize-0-1-9"
              scope="col"
            >
              cat
            </th>
            <th
              class="Table-th-0-1-4 zep-typo--normal-1 Table-capitalize-0-1-9"
              scope="col"
            >
              rental
            </th>
          </tr>
        </thead>
      </table>
    </div>
    <p
      class="Headline-text-0-1-1 zep-typo--normal-4"
    >
      Structure
    </p>
    <div
      class="Structure-root-0-1-11"
    >
      <img
        src="test-file-stub"
      />
    </div>
    <div
      class="Table-root-0-1-2"
    >
      <table
        class="Table-table-0-1-3"
      >
        <caption
          class="zep-visually-hidden"
        >
          structure
        </caption>
        <thead
          class="Table-header-0-1-7"
        >
          <tr
            class="Table-tr-0-1-5"
          >
            <th
              class="Table-th-0-1-4 zep-typo--normal-1 Table-capitalize-0-1-9"
              scope="col"
            >
              name
            </th>
            <th
              class="Table-th-0-1-4 zep-typo--normal-1 Table-capitalize-0-1-9"
              scope="col"
            >
              property
            </th>
            <th
              class="Table-th-0-1-4 zep-typo--normal-1 Table-lowercase-0-1-8"
              scope="col"
            >
              px
            </th>
            <th
              class="Table-th-0-1-4 zep-typo--normal-1 Table-lowercase-0-1-8"
              scope="col"
            >
              rem
            </th>
          </tr>
        </thead>
        <tbody
          class="Table-tbody-0-1-6"
        >
          <tr
            class="Table-tr-0-1-5"
          >
            <td
              class="Table-th-0-1-4 Table-borderBottom-0-1-10 zep-typo--normal-2"
              scope="row"
            >
              btn-height
            </td>
            <td
              class="Table-th-0-1-4 Table-borderBottom-0-1-10 zep-typo--normal-2"
            >
              height
            </td>
            <td
              class="Table-th-0-1-4 Table-borderBottom-0-1-10 zep-typo--normal-2"
            >
              48px
            </td>
            <td
              class="Table-th-0-1-4 Table-borderBottom-0-1-10 zep-typo--normal-2"
            >
              3rem
            </td>
          </tr>
          <tr
            class="Table-tr-0-1-5"
          >
            <td
              class="Table-th-0-1-4 Table-borderBottom-0-1-10 zep-typo--normal-2"
              scope="row"
            >
              btn-padding
            </td>
            <td
              class="Table-th-0-1-4 Table-borderBottom-0-1-10 zep-typo--normal-2"
            >
              padding
            </td>
            <td
              class="Table-th-0-1-4 Table-borderBottom-0-1-10 zep-typo--normal-2"
            >
              16px, 18px
            </td>
            <td
              class="Table-th-0-1-4 Table-borderBottom-0-1-10 zep-typo--normal-2"
            >
              1rem, 1.125rem
            </td>
          </tr>
        </tbody>
      </table>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllBySelectText": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getAllByValue": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getBySelectText": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "getByValue": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllBySelectText": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryAllByValue": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryBySelectText": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "queryByValue": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
